slice.go

```diff
@@ -742,3 +742,88 @@ func Splice[T any, Slice ~[]T](collection Slice, i int, elements ...T) Slice {
 
 	return append(append(append(output, collection[:i]...), elements...), collection[i:]...)
 }
+
+func verifyNonEmpty[T any](slice []T) {
+	if len(slice) == 0 {
+		panic("Cannot remove element from empty slice!")
```

Comment:
I don't really like calls to panic in such utilitary lib, any other idea?

Reply:
I think it should return bool value which indicates whether it is valid or not.